data/default.yaml

```diff
@@ -1,5 +1,5 @@
 ---
```

Comment:
shouldn't we add the correct setting for the merge behavior here? your target was to get `deep` as merge behavior?

@bastelfreak what whould be a good default? i personally prefer `first` over `deep` for pools. don't want my pools get collected from multiple files but for other persons `deep` is maybe the better default. so what to choose?

@sigv what would be your prefered default?

```
lookup_options:
  php::fpm::pools:
    merge: deep
```

Reply:
If the choice of merge strategy is so important, we probably need a note about this and how to change it in the `README.md` file.

Reply:
The option to choose other strategies is a standard Puppet feature. For example, `php::extensions` can already be configured to merge in different ways - it's also a Hash being taken from Hiera. The README does not have any documentation about changing merge behavior for that variable. Additionally, [automatic lookup documentation](https://puppet.com/docs/puppet/6.4/hiera_automatic.html#class-parameters) does cover the fact hashes are `first` by default:

> Note: Automatic lookup of class parameters uses the "first" merge method by default. You cannot change the default. If you want to get deep merges on keys, use the lookup_options feature.

Should I really add a note about this in this module's README in this PR?

Reply:
Just pushed a minor note in `README` as commit 5d59134 in this PR. Please check that out and let me know if further changes to this PR are of interest. Otherwise, I would prefer a _LGTM_ so that the change itself gets merged. 🙂 🙈

Reply:
readme looks good to me. 

i miss the following addition of ` data/default.yaml`:
```
lookup_options:
  php::fpm::pools:
    merge: first
```
i also would reference this file in the readme, so user see the context of the config. after that i would approve the MR

Reply:
👌🏻 Pushed `hash` as the override in `data/default.yaml` simply because `first` is already the default and overriding to it doesn't make that much sense. Added a link to the README pointing at `data/default.yaml` as well.

Reply:
can you please use `first`? my request to put it there was just because of having a reference what to change if someone want to change the behavior but i would prefere to stick with the default setting of `first`

Reply:
Done. It is set to `first` now in the local Hiera source.

Reply:
Ping.